### PR TITLE
Feature/api rating bookmark data model

### LINKED ISSUE #1130: Denormalize averageRating and bookmarksCount for API Backends
In order to sort the API Catalogue by average rating and bookmarks count, we may need to store the corresponding values in the ApiBackends collection. Create helpers and methods to calculate the average rating and bookmarks count. Make sure the helpers are called when a user bookmarks or rates an API Backend.

### PR DESCRIPTION
**Pull request by brylie**
Closes #1130 

# Proposed changes
- add `averageRating` and `bookmarkCount` fields to ApiBackends
- add `setAverageRating` collection helper for ApiBackends
- add `getAverageRating` and `getBookmarkCount` helpers for ApiBackends
- add `setAllApiBackendAverageRatings` method

### CODE REVIEW

File: catalogue/tabular.js [hunk 1]
```diff
@@ -17,9 +17,11 @@ TabularTables.ApiTable = new Tabular.Table({
     },
     {tmpl: Meteor.isClient && Template.favourite, title: "Bookmark"},
     {
-     tmpl: Meteor.isClient && Template.apiBackendRating,
-     title: "Rating"
-    }
+      data: "averageRating",
```
Comment (jykae) [inline catalogue/tabular.js]: Why is Tabular tables still hanging around here? You said to me that we switch it away and I have been creating table view on my PR. This does not make sense that you continued working with it and told me to do other way.

File: apis/server/methods/ratings.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    // Get all API Backends
+    const apiBackends = ApiBackends.find().fetch();
+
+    // Update the average rating value for each API Backend
+    apiBackends.forEach(function (apiBackend) {
+      // Set average rating value for current API Backend
+      apiBackend.setAverageRating();
+    });
+  },
+  setApiBackendAverageRating (apiBackendId) {
```
Comment (jykae) [inline apis/server/methods/ratings.js]: @brylie method name here could be something like set ApiBackendBookmarkCount

Comment (brylie) [inline apis/server/methods/ratings.js]: Which method? The `setApiBackendAverageRating` is related to the average rating field.

**Comment by jykae** [comment]
@brylie Overall nice work, other thing to fix before I merge this, 
we need to set new field values on probably on startup, possibly by doing it as migration (running once). Or any other idea for that?

File: apis/server/methods/ratings.js (hunk 2 above)
Comment (jykae) [inline apis/server/methods/ratings.js]: @brylie Ok, I confused this because console.log says about bookmarks. Could you also cleanup/fix console.logs?

**Comment by jykae** [comment]
Ok, merging in, so I can integrate this to my table view. Thanks @brylie

**Pull request merged**